opm/autodiff/SimulatorFullyImplicitBlackoilOutput.cpp

```diff
@@ -304,8 +304,15 @@ namespace Opm
         bool isIORank = output_ ;
         if( parallelOutput_ && parallelOutput_->isParallel() )
         {
+            // If this is not the initial write and no substep, then the well
+            // state used in the computation is actually the one of the last
+            // step. We need that well state for the gathering. Otherwise
+            // It an exception with a message like "global state does not
+            // contain well ..." might be thrown.
+            int wellStateStepNumber = ( ! substep && timer.reportStepNum() > 0) ?
+                (timer.reportStepNum() - 1) : timer.reportStepNum();
```

Comment:
@atgeirr I do not follow you. I do not change the number passed to writeTimeStep of the writer. `wellStateStepNumber` is only used to create the correct well state information for gathering the data. As the shut/open state of the wells is still the same as at the beginning of the report step we need to use that step and not the next one. In a sequential run this is done anyway as the state got initialized before the timer got incremented.